### PR TITLE
Adding at() method to Array, String & TypedArray

### PR DESCRIPTION
**Pull request by Rumyra**
A checklist to help your pull request get merged faster:
- [x] Summarize your changes

Adding known compat data for the relative indexing method: at() https://tc39.es/proposal-relative-indexing-method/

- [x] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)

Bugs for chrome & ff:
https://bugs.chromium.org/p/v8/issues/detail?id=10961
https://bugzilla.mozilla.org/show_bug.cgi?id=1681371

- [x] Data: if you tested something, describe how you tested with details like browser and version

Tested in FF Nightly & Chrome Canary

- [x] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [x] Link to related issues or pull requests, if any

https://github.com/mdn/content/issues/307

### CODE REVIEW

**Comment by Rumyra** [comment]
Linter doesn't like the spec url (which I added the proposal for) - without this being moved to next stage and added to the spec I'm unsure how to proceed.

**Comment by Elchi3** [comment]
> Linter doesn't like the spec url (which I added the proposal for) - without this being moved to next stage and added to the spec I'm unsure how to proceed.

It wants you to provide a deep link to the feature, e.g. https://tc39.es/proposal-relative-indexing-method/#sec-array.prototype.at (and for string and typed array respectively)

Unfortunately, the error messages the schema validator produces aren't that helpful here. spec_url is documented here: https://github.com/mdn/browser-compat-data/blob/master/schemas/compat-data-schema.md#the-__compat-object

File: javascript/builtins/Array.json [hunk 1]
```diff
@@ -105,6 +105,58 @@
...
+                "version_added": "89"
+              },
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "firefox": {
+                "version_added": "85"
```
Comment (evilpie) [inline javascript/builtins/Array.json:123]: I think per convention this needs to be false, because it's only enabled in Firefox Nightly.

Comment (Rumyra) [reply javascript/builtins/Array.json:123]: I'm going to delve a little deeper as the chrome version added is canary, so the same rules should apply one assumes... I'll read the guidelines ☺️

**Comment by Rumyra** [comment]
Thank you @Elchi3 - I've fixed the urls and also there was a order issue with typedarr 👍

**Comment by chrisdavidmills** [comment]
@Rumyra @Elchi3 just to check — is this ready to be merged?

File: javascript/builtins/String.json [hunk 2]
```diff
@@ -158,6 +158,58 @@
...
+                "version_added": "89"
+              },
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "firefox": {
+                "version_added": "85"
```
Comment (Elchi3) [inline javascript/builtins/String.json:176]:
```suggestion
                "version_added": false
```

File: javascript/builtins/String.json [hunk 3]
```diff
@@ -158,6 +158,58 @@
...
+              },
+              "firefox_android": {
+                "version_added": "85"
```
Comment (Elchi3) [inline javascript/builtins/String.json:179]:
```suggestion
                "version_added": false
```

File: javascript/builtins/TypedArray.json [hunk 4]
```diff
@@ -104,6 +104,58 @@
...
+                "version_added": "89"
+              },
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "firefox": {
+                "version_added": "85"
```
Comment (Elchi3) [inline javascript/builtins/TypedArray.json:122]:
```suggestion
                "version_added": false
```

File: javascript/builtins/TypedArray.json [hunk 5]
```diff
@@ -104,6 +104,58 @@
...
+              },
+              "firefox_android": {
+                "version_added": "85"
```
Comment (Elchi3) [inline javascript/builtins/TypedArray.json:125]:
```suggestion
                "version_added": false
```

File: javascript/builtins/Array.json [hunk 6]
```diff
@@ -105,6 +105,58 @@
...
+        "at": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/at",
+            "spec_url": "https://tc39.es/proposal-relative-indexing-method/#sec-array-prototype-additions",
+            "support": {
+              "chrome": {
+                "version_added": "89"
+              },
+              "chrome_android": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/Array.json:117]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/Array.json [hunk 7]
```diff
@@ -105,6 +105,58 @@
...
+              },
+              "edge": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/Array.json:120]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/Array.json [hunk 8]
```diff
@@ -105,6 +105,58 @@
...
+              },
+              "safari_ios": {
+                "version_added": false
+              },
+              "samsunginternet_android": {
+                "version_added": false
+              },
+              "webview_android": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/Array.json:150]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/String.json [hunk 9]
```diff
@@ -158,6 +158,58 @@
...
+        "at": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String/at",
+            "spec_url": "https://tc39.es/proposal-relative-indexing-method/#sec-string-prototype-additions",
+            "support": {
+              "chrome": {
+                "version_added": "89"
+              },
+              "chrome_android": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/String.json:170]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/String.json [hunk 10]
```diff
@@ -158,6 +158,58 @@
...
+              },
+              "edge": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/String.json:173]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/String.json [hunk 11]
```diff
@@ -158,6 +158,58 @@
...
+              },
+              "safari_ios": {
+                "version_added": false
+              },
+              "samsunginternet_android": {
+                "version_added": false
+              },
+              "webview_android": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/String.json:203]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/TypedArray.json [hunk 12]
```diff
@@ -104,6 +104,58 @@
...
+        "at": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/at",
+            "spec_url": "https://tc39.es/proposal-relative-indexing-method/#sec-%typedarray.prototype%-additions",
+            "support": {
+              "chrome": {
+                "version_added": "89"
+              },
+              "chrome_android": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/TypedArray.json:116]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/TypedArray.json [hunk 13]
```diff
@@ -104,6 +104,58 @@
...
+              },
+              "edge": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/TypedArray.json:119]:
```suggestion
                "version_added": "89"
```

File: javascript/builtins/TypedArray.json [hunk 14]
```diff
@@ -104,6 +104,58 @@
...
+              },
+              "safari_ios": {
+                "version_added": false
+              },
+              "samsunginternet_android": {
+                "version_added": false
+              },
+              "webview_android": {
+                "version_added": false
```
Comment (Elchi3) [inline javascript/builtins/TypedArray.json:149]:
```suggestion
                "version_added": "89"
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]

File: javascript/builtins/Array.json (hunk 1 above)
Comment (Elchi3) [reply javascript/builtins/Array.json:123]: Firefox does not ship to release, so yes, it should be false there.

Chrome versions are fine as https://www.chromestatus.com/feature/6123640410079232 says it is enabled by default and thus will ride these train to release.

Comment (Rumyra) [reply javascript/builtins/Array.json:123]: I am happy to make these changes @Elchi3 as I am unsure as to what we add to BCD as far as future versions go... however at() is also enabled by default in FF85 - does it not follow the same rules?

Comment (Elchi3) [reply javascript/builtins/Array.json:123]: > however at() is also enabled by default in FF85 - does it not follow the same rules?

So, https://bugzilla.mozilla.org/show_bug.cgi?id=1681371 is not yet resolved fixed, and so I think it is not yet enabled by default.

> I am unsure as to what we add to BCD as far as future versions go...

Once https://bugzilla.mozilla.org/show_bug.cgi?id=1681371 is implemented in Firefox we will have a version number we can use.

Comment (Rumyra) [reply javascript/builtins/Array.json:123]: Thank you @Elchi3 🙏 the more I know...

File: javascript/builtins/String.json [hunk 15]
```diff
@@ -158,6 +158,58 @@
...
+              },
+              "webview_android": {
+                "version_added": "89"
+              }
+            },
+            "status": {
+              "experimental": true,
+              "standard_track": true,
+              "deprecated": true
```
Comment (Elchi3) [inline javascript/builtins/String.json:209]:
```suggestion
              "deprecated": false
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Just one more thing then this looks excellent :)
Thanks for your work, @Rumyra 🙏

File: javascript/builtins/String.json (hunk 15 above)
Comment (Rumyra) [reply javascript/builtins/String.json:209]: 🤦🏻‍♀️ I'm so glad you're reviewing @Elchi3 - I feel I've learnt a lot here today ☺️

**Review by Elchi3 — APPROVED** [review]
Awesome, let's merge these fancy new methods! 👍

**Pull request merged**